core/os/device/deviceinfo/cc/vk.cpp

```diff
@@ -163,6 +166,60 @@ bool vkPhysicalDevices(
         std::string(prop.deviceName));
   }
 
+  uint32_t queue_family_index = static_cast<uint32_t>(-1);
+  VkPhysicalDevice physical_device{};
```

Comment:
My other comment got eaten.
We probably want to create a VkDevice for EVERY physical device, instead of simple "one physical device at random" that way if there is also something like swiftshader on the system, we don't actually create the wrong one.